src/ppxgboost/PPTree.py

```diff
@@ -269,12 +269,8 @@ def parse_subtree(s, lvl):
     number_regex = r'-?(\d*\.\d+|\d+)(e-?\d+)?'
 
     # a regex for parsing feature names
-    # '\w' means find all word characters - e.g. matches a "word" character: a letter or digit
-    # or underbar [a-zA-Z0-9_] (Note that \w contains underscore)
-    # '\s' means whitespace
-    # '.' and '-' are literal characters
-    # Thus, a feature name is a sequence of one or more of these characters.
-    feature_name_regex = r'[\w\s.-]+'
+    # This reads all characters lazily until the delimiter for the feature name '<' is reached
+    feature_name_regex = r'.+?'
```

Comment:
If you want to read all characters until you get a '<', shouldn't this be r'[^<]*'?

Reply:
Updated